Main updates

ACC01ADE opened the pull request:
## Describe Changes

I made this more better by doing ...

- Implementing ERC20 tokens
- Testing new standards enforcement process


## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] Code styles have been enforced

ACC01ADE committed:
version control

ACC01ADE committed:
updates

ACC01ADE committed:
minor updates

ACC01ADE committed:
updates

ACC01ADE committed:
ERC20 improvements

ACC01ADE committed:
Implementing eip-4524

ACC01ADE committed:
Upgraded to Solidity v0.8.13

sogoiii commented on contracts/Holograph.sol:
```diff
@@ -101,7 +101,7 @@
 */
 
-pragma solidity 0.8.11;
+pragma solidity 0.8.13;
```
@thisisvitto you mentioned that 0.8.13 was failing. This should be 0.8.11 right? Lets isolate updating the solidity version into its own task.

ACC01ADE replied:
For me it's important that minor version differences don't affect the overall code's functionality.
With latest work, was able to test between v0.8.11, v0.8.12, and v0.8.13 with success.
Now can leave it alone at v0.8.13 for the foreseeable future.

sogoiii commented on .nvmrc:
```diff
@@ -1 +1 @@
-v16.6.1
\ No newline at end of file
+v16.14.2
```
Since we are using asdf now we can remove .nvmrc

ACC01ADE replied:
Updated

sogoiii reviewed (commented):
So when i run `_deploy.sh` it stops in `deploy/bridge_1...` with `Deployed MultichainERC721 Contract: 0x36648C73AeB836Cb93369D1B7202ADC1C219405b`

so it does not work for me locally.

I see that there are binaries to the different solidity version. That is not necessary.

ACC01ADE committed:
hToken

ACC01ADE committed:
minor cleanup/re-org

ACC01ADE committed:
updating ABI

ACC01ADE committed:
Updated tokens to enforce owner only functions.\
Updated deployment pipepline to specify deployer as owner.\
Updated deployment tools to allow for multiple wallets per provider, and disabled nonce-sharing between providers

ACC01ADE committed:
Adding hToken testing

ACC01ADE committed:
Adding mock ERC20 tokens
Updated README file.
Changed deploy naming convention to test, for more clarity

ACC01ADE commented:
Can you try to run the latest version to see if it works for you now? @sogoiii

ACC01ADE committed:
removing .nvmrc in favor of asdf

ACC01ADE committed:
Improved the README.md instructions.

ACC01ADE committed:
Improved the README.md instructions.

ACC01ADE committed:
cloc

ACC01ADE committed:
Minor improvements
Additional hToken testing
hToken bug fixes

ACC01ADE committed:
Adding ERC20 support to bridge

ACC01ADE committed:
Implementing sample CXIP-ERC721 smart contract

ACC01ADE committed:
adding Hardhat

sogoiii committed:
set default of solidity version.

sogoiii reviewed (approved):
good progress. Merging to move code forward and make the next PRs easier.
